src/pt/sussyscan/src/eu/kanade/tachiyomi/extension/pt/sussyscan/SussyScan.kt

```diff
@@ -1,34 +1,207 @@
 
-    override val useLoadMoreRequest = LoadMoreStrategy.Never
-    override val useNewChapterEndpoint = true
+    override fun headersBuilder() = super.headersBuilder()
+        .set("scan-id", "1")
+
+    // ============================= Popular ==================================
+
+    override fun popularMangaRequest(page: Int) = GET("$apiUrl/obras/top5", headers)
+
+    override fun popularMangaParse(response: Response): MangasPage {
+        val dto = response.parseAs<WrapperDto<List<MangaDto>>>()
+        val mangas = dto.results.map(::toSManga)
+        return MangasPage(mangas, dto.hasNextPage())
+    }
+
+    private fun toSManga(dto: MangaDto) = SManga.create().apply {
+        title = dto.name
+        thumbnail_url = dto.thumbnail
+        Jsoup.parseBodyFragment(dto.description).let { description = it.text() }
+        url = "/obras/${dto.id}/${dto.slug}#${dto.scanId}"
+        initialized = true
+        status = dto.status.toStatus()
+    }
+
+    // ============================= Latest ===================================
+
+    override fun latestUpdatesRequest(page: Int): Request {
+        val url = "$apiUrl/obras/novos-capitulos".toHttpUrl().newBuilder()
+            .addQueryParameter("pagina", page.toString())
+            .addQueryParameter("limite", "24")
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun latestUpdatesParse(response: Response) = popularMangaParse(response)
+
+    // ============================= Search ===================================
+
+    override fun searchMangaRequest(page: Int, query: String, filters: FilterList): Request {
+        val url = "$apiUrl/obras".toHttpUrl().newBuilder()
+            .addQueryParameter("pagina", page.toString())
+            .addQueryParameter("limite", "8")
+            .addQueryParameter("obr_nome", query)
+            .build()
+        return GET(url, headers)
+    }
+
+    override fun searchMangaParse(response: Response) = popularMangaParse(response)
+
+    // ============================= Details ==================================
+
+    override fun getMangaUrl(manga: SManga) = "$baseUrl${manga.url}"
+        .replace("obras", "obra")
+
+    override fun mangaDetailsRequest(manga: SManga) =
+        GET("$apiUrl${manga.url.substringBeforeLast("/")}", headers)
+
+    override fun mangaDetailsParse(response: Response) =
+        toSManga(response.parseAs<WrapperDto<MangaDto>>().results)
 
-    override val mangaDetailsSelectorTitle = "${super.mangaDetailsSelectorTitle}, span.rate-title, title"
-    override val mangaDetailsSelectorThumbnail = "head meta[property='og:image']"
+    // ============================= Chapters =================================
 
-    override fun mangaDetailsParse(document: Document) = super.mangaDetailsParse(document).apply {
-        title = title.substringBeforeLast("–")
+    override fun getChapterUrl(chapter: SChapter) = "$baseUrl${chapter.url}"
+        .replace("capitulos", "capitulo")
```

Comment:
Might cause issues? Use `HttpUrl`, check if pathSegment at index x, iff y: set to z.
x = index
y = search
z = replacement